### DISCUSSION TITLE
Notifications Tab Context/Features

### DISCUSSION THREAD

**Discussion by cryptoKevinL** [MetaMask/snaps#1476] [Q&A] [answered]
I am writing a Snap to enable unread message notifications, extending our wallet-to-wallet chat app from WalletChat.fun.  I have new unread notifications working, I am just curious if there is any additional context or actions that can happen from clicking a notification?

1) Is there any plan to allow to users to click a notification, and allow Snap devs to take action on that (pop up an already/dialog so a user could respond in-app or see additional message details)
<img width="299" alt="Screenshot 2023-06-04 at 8 46 21 AM" src="https://github.com/MetaMask/snaps/assets/19207330/3ac48ed5-5c6a-4e61-a888-3e70de802a6b">

2) is there any plan to allow Snaps devs to use the browser extension alert/notification count for custom purposes? 
<img width="49" alt="Screenshot 2023-06-03 at 12 39 09 PM" src="https://github.com/MetaMask/snaps/assets/19207330/17e142de-29ff-40b1-b7d9-1ef5690af9f0">

For example, Unread notifications from our WalletChat snap, and when a user opens the icon it they would have an alert waiting (not as intrusive as a pop-up but it shows the user that something is waiting there)

Still a great addition to our dApp as it stands, but wanted to bring these up in case I missed features I could already be using, or other have ideas using other UI component available to us.  Thanks!

**Comment by Montoya** [comment, marked as answer]
1. We have plans to implement such actions as part of our "notifications v2" roadmap. These improvement will come after the first launch of Snaps to the extension. 
2. No, that won't be possible.

**Comment by cryptoKevinL** [comment]
Thanks - looking forward to v2!  Awesome work so far - everything works great and the examples are super helpful.
